src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -2,88 +2,180 @@
-                        goaLines.add(generateGOCellularCompartmentLine(proteinInst, reactionInst, taxonIdentifier));
-                    } else {
-                        logger.info("Species has an alternative GO compartment, skipping GO annotation");
-                    }
-                } else {
-                    logger.info("Protein is from an excluded microbial species, skipping GO annotation");
+            String issueDisqualifyingProtein = getAnyIssueForCellularComponentDisqualification(protein);
+            if (issueDisqualifyingProtein.isEmpty()) {
+                String goaLine = generateGOCellularCompartmentLine(protein, reactionlikeEvent);
+                if (!goaLine.isEmpty()) {
+                    goaLines.add(goaLine);
                 }
             } else {
-                logger.info("Invalid protein, skipping GO annotation");
+                logger.warn(issueDisqualifyingProtein);
             }
         }
         return goaLines;
     }
 
     /**
-     * Proteins reaching this part of the script are deemed valid for GO annotation. Retrieves the Cellular Compartment accession associated with this instance and then calls the GOA line generator.
-     * @param proteinInst -- GKInstance, Individual protein instance from the retrieved proteins.
-     * @param reactionInst -- GKInstance, Parent reaction that the protein is associated with.
-     * @param taxonIdentifier -- String, CrossReference ID of protein's species.
+     * Performs an AttributeQueryRequest on the incoming reaction instance. This will retrieve all protein's
+     * affiliated with the Reaction.
+     * @param reactionlikeEvent -- GKInstance from ReactionlikeEvent class.
+     * @return -- Set of GKInstances output from the AttributeQueryRequest.
      * @throws Exception -- MySQLAdaptor exception.
      */
-    private static String generateGOCellularCompartmentLine(GKInstance proteinInst, GKInstance reactionInst, String taxonIdentifier) throws Exception {
-        String reactomeIdentifier = REACTOME_IDENTIFIER_PREFIX  + GOAGeneratorUtilities.getStableIdentifierIdentifier(reactionInst);
-        GKInstance referenceEntityInst = (GKInstance) proteinInst.getAttributeValue(ReactomeJavaConstants.referenceEntity);
-        String goCellularCompartmentAccession = getCellularCompartmentGOAccession(proteinInst);
-        String goaLine = "";
-        if (!goCellularCompartmentAccession.isEmpty()) {
-            goaLine = GOAGeneratorUtilities.generateGOALine(referenceEntityInst, CELLULAR_COMPONENT_LETTER, goCellularCompartmentAccession, reactomeIdentifier, TRACEABLE_AUTHOR_STATEMENT_CODE, taxonIdentifier);
-            GOAGeneratorUtilities.assignDateForGOALine(proteinInst, goaLine);
-        } else {
-            logger.info("Protein has no Cellular Compartment accession, skipping GO annotation");
+    static Set<GKInstance> retrieveProteins(GKInstance reactionlikeEvent) throws Exception {
```

Comment:
Scope is "package" but this method is only referenced in the class where it is declared. This method could be declared as private (unless there's a good reason to make it package-scoped).

Reply:
Yup, good catch - reworked to be private.